lib/DBDefs/Default.pm

```diff
@@ -192,6 +192,26 @@ sub GOOGLE_CUSTOM_SEARCH { '' }
 # you use it in your own definitions.
 sub CACHE_NAMESPACE { 'MB:' }
 
+# Redis by default has 16 numbered databases available, of which DB 0 is the
+# default. Here you can configure which of these databases are used by
+# musicbrainz-server. It is not recommended to change the defaults unless
+# you are sharing a Redis instance with other services. (We do not advise
+# this; please heed the note about `REDIS_TEST_DATABASE` below, or suffer
+# data loss.)
+#
+# We allow configuring both the DB numbers used by the cache (for entities)
+# and the store (for sessions), as these may point to different Redis
+# instances.
+#
+# `REDIS_TEST_DATABASE` indicates the minimum DB number used when running
+# tests. This is used for both the cache and the store. Note that the tests
+# may use additional DB numbers greater than or equal to
+# `REDIS_TEST_DATABASE`, and that all production DB numbers should be less
+# than it. All test databases will be completely erased on each test run.
+sub REDIS_CACHE_DATABASE { 0 }
+sub REDIS_STORE_DATABASE { 0 }
```

Comment:
That was in commit https://github.com/metabrainz/musicbrainz-server/commit/4ce0c3563e706f90bd02873a464407bb16867a37 which has been superseded with commit c34c72214b08bfe8234b9330fa470ebf2aa29a33.

Is it needed to specify `database => 0` if it is the default value in Redis?

Reply:
> Do both cache and store have database 0 because these are hold by different Redis instances? What about mirror which uses only one Redis instance? What would be the inconvenience of setting 1 to store?

 * The cache and store don't have to be separate in order to use 0 for both: we also use different cache prefixes to avoid collisions, so they can both live in database 0 in the same instance.
 * I otherwise assume 0 is only used for both because it's the default.
 * I'm not sure it would necessarily be an inconvenience, though when connecting with `redis-cli` you'll have to pass the `-n` flag to select the appropriate database number.

> That was in commit https://github.com/metabrainz/musicbrainz-server/commit/4ce0c3563e706f90bd02873a464407bb16867a37 which has been superseded with commit https://github.com/metabrainz/musicbrainz-server/commit/c34c72214b08bfe8234b9330fa470ebf2aa29a33.

I removed the `REDIS_CACHE_DATABASE` and `REDIS_STORE_DATABASE` flags because existing DBDefs files (which are typically based on DBDefs.pm.sample) would not even reference them at all.

> Is it needed to specify database => 0 if it is the default value in Redis?

It shouldn't be, though I added it as a form of documentation (to indicate that it's a property which can be set).

Reply:
I will add, though, that storing the sessions in DB1 would have a benefit in that you could easily clear the cache without dropping login sessions. (That's only a problem if you're sharing an instance.)